flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  description =
+    "Tool to simplify secure shell connections over AWS Systems Manager (SSM).";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?ref=nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        awssh = pkgs.python310Packages.buildPythonPackage rec {
+          pname = "awssh";
+          version = "1.1.0";
+          doCheck = false;
+          propagatedBuildInputs = with pkgs.python310Packages; [
+            boto3
+            docopt
+            schema
+            setuptools
+            wheel
+          ];
+          src = pkgs.fetchFromGitHub {
+            owner = "cisagov";
+            repo = "awssh";
+            rev = "v1.1.0";
+            sha256 = "sha256-4b2VBFUQye4wTvuagPwEImLwkUO4Dk5hvOYW+eg8OGA=";
+          };
+        };
+      in {
+        packages.default = pkgs.python310Packages.buildPythonPackage rec {
+          pname = "awssh";
+          version = "1.1.0";
```

Comment:
Is this necessary? It would be nice if it could just use whatever version is defined in the package.

Reply:
This is related to @dav3r's comment [here](https://github.com/cisagov/awssh/pull/27#discussion_r1303359341).

Reply:
Yes, I've updated the other reference to use a variable though Please see cd5e9844ea703968202363f8a04ef49839a2693a

>It would be nice if it could just use whatever version is defined in the package.
But this would be on a release basis wouldn't it?

Reply:
@jasonodoom - Please see [here](https://github.com/cisagov/skeleton-docker/blob/develop/bump_version.sh) for an example of the way we update the `bump_version.sh` script to keep "extra" files like this in sync with the project's version.  This is what @mcdonnnj and @dav3r are trying to point you toward.

Reply:
@jsf9k @dav3r Are you okay with me implementing an action to update the `flake.lock` file as a result of this change? https://github.com/DeterminateSystems/update-flake-lock

Reply:
@jasonodoom I'm not following- what does adding that action have to do with contents of this conversation?

Reply:
@dav3r you want the version number updated but `flake.lock` needs to be rebuilt as a result of this. Simply changing the file to update the version number isn't a good practice without updating `flake.lock`

Reply:
Got it, now I am tracking.  The `flake.lock` needs to be updated for more than just version updates though.  As long as it gets updated whenever it's supposed to, I'm fine with whatever mechanism you choose to do that.

Reply:
@dav3r and I discussed this and this will be added to a TODO.

Reply:
I think the update of the `flake.lock` file can be done in the `bump_version.sh` script.  If that means that folks need to install Nix locally in order to wrangle this repository then that is fine.  If that is the case then there should be a mention of this requirement in the `README.md` file as well.

Reply:
@jsf9k @dav3r please see https://github.com/cisagov/awssh/pull/27/commits/4cef09be43daa61d8b72773fd0d98b14d78d81e5

Reply:
@jasonodoom Can you please do the following:

1. Use a variable for that `flake.nix` file, like what we do for [the version file](https://github.com/cisagov/awssh/blob/550a17d9d8dda7fcd4b523014a9344938a34a6eb/bump_version.sh#L9)
2. Add a comment explaining why we are running `nix flake update`

Reply:
@dav3r Please see https://github.com/cisagov/awssh/pull/27/commits/22d2bf270009fafafb74111ddb45a573d4eaefed

Reply:
One more thing: https://github.com/cisagov/awssh/pull/27/commits/be22e876aef6490796c689e6f5e7dd8406f18bf4

Reply:
I believe the `sed` logic needs to be duplicated to the `finalize` command. The `nix flake update` logic should specifically **not** be duplicated.

Reply:
@mcdonnnj - Are you good with resolving this conversation?  @jasonodoom has taken care of the items that I noted.

Reply:
@mcdonnnj Please see https://github.com/cisagov/awssh/pull/27/commits/782d54fa0c3b57386fc7b4860113d6e0959e3bc1

Reply:
That works! Thanks!